ecommerce/api.py

```diff
@@ -480,3 +482,36 @@ def calculate_run_price(course_run, user):
     coupon = coupons[0]
     price = calculate_coupon_price(coupon, price, course_run.edx_course_key)
     return price, coupon
+
+
+def validate_prices():
+    """
+    Validate prices and financial aid discounts
+    Returns:
+        list: List of validation errors
+    """
+    errors = []
+    programs = Program.objects.filter(live=True)
+    for program in programs:
+        runs = CourseRun.objects.filter(course__program=program)
+        course_prices = [CoursePrice.objects.filter(course_run=run, is_valid=True).first() for run in runs]
+        prices = [course_price.price if course_price else None for course_price in course_prices]
+
+        if not (all(prices[0] == price for price in prices) and prices[0]):
```

Comment:
What is `prices[0]` checking for here? If `prices` is an empty list this will cause a `IndexError`, and we don't want to do anything different if the price is `0` or if it's greater than `0`